runtime/src/iree-amd-aie/driver/xrt/xrt_device.cc

```diff
@@ -104,7 +117,9 @@ static void iree_hal_xrt_device_destroy(iree_hal_device_t* base_device) {
 
   iree_hal_allocator_release(device->device_allocator);
   iree_arena_block_pool_deinitialize(&device->block_pool);
+  xrtDeviceHandle device_hdl = device->device_hdl;
   iree_allocator_free(host_allocator, device);
+  (void)xrtDeviceClose(device_hdl);
```

Comment:
The way `xrtDeviceOpen` and `xrtDeviceClose` work is there's a `static std::map<...>` in XRT that maintains "references" to the [device objects](https://github.com/nod-ai/XRT/blob/a01266e20dff0ec1f59059a3377102a12e2c7d89/src/runtime_src/core/common/api/xrt_device.cpp#L46).